### PR TITLE
Organize game code

### PR DESCRIPTION
**Pull request by neivv**
No functionality changes, just modifying how we interact with BW

### CODE REVIEW

File: game/src/bw_1161.rs [hunk 1]
```diff
@@ -0,0 +1,612 @@
...
+    }
+}
+
+unsafe fn create_lobby(
+    map_path: &Path,
+    name: &str,
+    game_type: bw::GameType,
+) -> Result<(), bw::LobbyCreateError> {
+    let map = find_map_entry(map_path)?;
+    // Password must be null for replays to work
```
Comment (2Pacalypse-) [inline game/src/bw_1161.rs]: Is this correct? I thought when watching replays through a lobby, password must *not* be null?
Or is this related to something else?

**Comment by 2Pacalypse-** [comment]
Btw, does this bit in description still holds?
>The code of the game DLL has two major parts: asynchronous code that focuses on handling network traffic, and 'synchronous' code that executes inside of hooks into BW's code. They currently are mixed in src/ directory, but it may be worth separating one/both of them in separate modules, even though most of the files aren't 100% async or 100% BW.

**Comment by neivv** [comment]
It kind of still holds? 1.16.1-specific code is now under bw_1161/, and that's obviously almost all of bw-side code, but there's still bunch of files whose code is run on bw's thread. I'll revise the documentation a bit once scr code is added =)

File: game/src/bw_1161.rs (hunk 1 above)
Comment (neivv) [reply game/src/bw_1161.rs]: You cannot set password anyway in LAN, but for some reason hosting replay will specifically check that there isn't password.. I think? Anyway it's been like that always here

**Comment by 2Pacalypse-** [comment]
Alright, sounds good. As far as I'm concerned you can merge this into the master. "Squash and merge" option on github should probably be used here.
